### PR TITLE
Migrate "actions/boost-versions" CI to GA

### PR DESCRIPTION
**Pull request by nikita-bykov**
In scope of this PR we've moved boost-versions CI from AzureDevOps to GitHub Actions

### CODE REVIEW

File: .github/workflows/build-boost-packages.yml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        if ('${{ matrix.platform }}' -eq 'win32') { powershell ./setup.ps1 } else { sh ./setup.sh }
+      working-directory: ${{ runner.temp }}/${{ env.ARTIFACT_NAME }}
+
+    - name: Setup BOOST_ROOT ${{ github.event.inputs.VERSION }}
+      run: |
+        $targetPath = $env:AGENT_TOOLSDIRECTORY
+        if ([string]::IsNullOrEmpty($targetPath)) {
+            # GitHub Windows images don't have `AGENT_TOOLSDIRECTORY` variable
+            $targetPath = $env:RUNNER_TOOL_CACHE
+        }
```
Comment (maxim-lobanov) [inline .github/workflows/build-boost-packages.yml:121]: We should move this logic to setup script

File: .github/workflows/build-boost-packages.yml [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    - name: Publish Release ${{ env.VERSION }}
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ env.VERSION }}-${{ github.run_id }}
+        release_name: ${{ env.VERSION }}
+        body: |
+          Boost ${{ env.VERSION }}
```
Comment (maxim-lobanov) [inline .github/workflows/build-boost-packages.yml:156]: empty line between steps

File: builders/boost-builder.psm1 [hunk 3]
```diff
@@ -75,6 +78,10 @@ class BoostBuilder {
         Generates Boost artifact from downloaded binaries.
         #>
 
+        Write-Host "Create WorkFolderLocation and ArtifactFolderLocation folders"
+        New-Item -Path $this.WorkFolderLocation -ItemType "directory"
```
Comment (maxim-lobanov) [inline builders/boost-builder.psm1:82]: `Directory`

File: tests/Windows.Tests.ps1 [hunk 4]
```diff
@@ -24,7 +21,7 @@ Describe "Windows Tests" {
             "/EHsc",
             "/I", "${env:BOOST_ROOT}\include",
             "main-headers.cpp",
-            "/link", "/LIBPATH:${env:BOOST_ROOT}\lib",
+            "/link", "/LIBPATH:${env:BOOST_ROOT}\lib"
```
Comment (maxim-lobanov) [inline tests/Windows.Tests.ps1:24]: Why we remove `,` here?

File: .github/workflows/build-boost-packages.yml [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+            # GitHub Windows images don't have `AGENT_TOOLSDIRECTORY` variable
+            $targetPath = $env:RUNNER_TOOL_CACHE
+        }
+        $BoostToolcachePath = Join-Path -Path $targetPath -ChildPath "boost"
+        $BoostToolcacheVersionPath = Join-Path -Path $BoostToolcachePath -ChildPath "${{ github.event.inputs.VERSION }}"
+        $boostDirectory = Join-Path $BoostToolcacheVersionPath "${{ matrix.architecture }}"
+        $LD_LIBRARY = Join-Path -Path $boostDirectory -ChildPath "lib"
+        Write-Host "BOOST_ROOT = ${boostDirectory}"
+        echo "::set-env name=BOOST_ROOT::${boostDirectory}"
+        echo "::set-env name=LD_LIBRARY_PATH::${LD_LIBRARY}"
```
Comment (maxim-lobanov) [inline .github/workflows/build-boost-packages.yml:128]: Can we move this logic to Test file?

File: .github/workflows/build-boost-packages.yml (hunk 2 above)
Comment (nikita-bykov) [reply .github/workflows/build-boost-packages.yml:157]: Added the empty line

File: builders/boost-builder.psm1 (hunk 3 above)
Comment (nikita-bykov) [reply builders/boost-builder.psm1]: Fixed it

File: tests/Windows.Tests.ps1 (hunk 4 above)
Comment (nikita-bykov) [reply tests/Windows.Tests.ps1]: Returned it

**Review by maxim-lobanov — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Pull request merged**